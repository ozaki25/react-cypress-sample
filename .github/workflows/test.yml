name: Test
on: [pull_request]
jobs:
  unit-test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - name: Use Node.js 12
        uses: actions/setup-node@v1
        with:
          node-version: 12
      - name: install
        run: |
          yarn install
      - name: lint
        run: |
          yarn lint
      - name: test
        run: |
          yarn test
  e2e-test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - name: Use Node.js 12
        uses: actions/setup-node@v1
        with:
          node-version: 12
      - name: install
        run: |
          yarn install
      - name: e2e-test
        run: |
          yarn e2e
