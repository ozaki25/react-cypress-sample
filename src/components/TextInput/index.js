import React from 'react';
import styled from 'styled-components';

const Input = styled.input`
  font-size: 1.2em;
  padding: 0.3em;
  border: none;
  border-bottom: 1px solid #9e9e9e;
  width: 1.8em;
  text-align: center;
`;

function TextInput(props) {
  return <Input {...props} />;
}

export default TextInput;
