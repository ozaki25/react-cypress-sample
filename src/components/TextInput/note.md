- 入力域のコンポーネントです
- 数字2文字を入れる想定なので幅は狭いです