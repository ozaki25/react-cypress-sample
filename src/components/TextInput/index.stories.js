import React from 'react';
import { action } from '@storybook/addon-actions';
import TextInput from '.';
import note from './note.md';

export default {
  title: 'TextInput',
  parameters: {
    info: {
      text: note,
    },
  },
};

export const nomal = () => <TextInput onChange={action('changed')} />;
