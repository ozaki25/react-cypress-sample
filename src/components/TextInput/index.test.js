import React from 'react';
import { snapshot } from '../../test/helper';
import TextInput from '.';

snapshot('TextInput', <TextInput />);
