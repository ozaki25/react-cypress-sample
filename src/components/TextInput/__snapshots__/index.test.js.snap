// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: TextInput 1`] = `
.c0 {
  font-size: 1.2em;
  padding: 0.3em;
  border: none;
  border-bottom: 1px solid #9e9e9e;
  width: 1.8em;
  text-align: center;
}

<input
  className="c0"
/>
`;
